packages/utils/src/types.ts

```diff
@@ -282,8 +276,17 @@ export type RegistryWidgetsType<T = any, S extends StrictRJSFSchema = RJSFSchema
   [name: string]: Widget<T, S, F>;
 };
 
+export type RJSFBaseProps<T = any, S extends StrictRJSFSchema = RJSFSchema, F extends FormContextType = any> = {
+  /** The schema object for the field being described */
+  schema: S;
+  /** The uiSchema object for this description field */
+  uiSchema?: UiSchema<T, S, F>;
+  /** The `registry` object */
+  registry: Registry<T, S, F>;
+};
+
```

Comment:
I like this utility type, and...

Can you move this up before its first usage?